bundles/org.openhab.binding.myenergi/src/main/java/org/openhab/binding/myenergi/internal/handler/MyenergiBaseDeviceHandler.java

```diff
@@ -0,0 +1,300 @@
+ */
+package org.openhab.binding.myenergi.internal.handler;
+
+import static org.openhab.core.thing.ThingStatus.ONLINE;
+
+import java.time.ZonedDateTime;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.measure.Unit;
+import javax.measure.quantity.Dimensionless;
+import javax.measure.quantity.ElectricPotential;
+import javax.measure.quantity.Energy;
+import javax.measure.quantity.Frequency;
+import javax.measure.quantity.Power;
+import javax.measure.quantity.Temperature;
+import javax.measure.quantity.Time;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.myenergi.internal.MyenergiDeviceConfiguration;
+import org.openhab.binding.myenergi.internal.exception.ApiException;
+import org.openhab.core.cache.ExpiringCache;
+import org.openhab.core.config.core.status.ConfigStatusCallback;
+import org.openhab.core.config.core.status.ConfigStatusSource;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingStatusInfo;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link MyenergiBaseDeviceHandler} is an abstract base class for any
+ * MyEnergi things. It contains an expiring
+ * cache which ensures thing properties are not unnecessarily updated.
+ *
+ * @author Rene Scherer - Initial Contribution
+ */
+@NonNullByDefault
+public abstract class MyenergiBaseDeviceHandler extends BaseThingHandler implements ConfigStatusCallback {
+
+    private static final int UPDATE_THING_CACHE_TIMEOUT = 3000; // 3 secs
+
+    private final Logger logger = LoggerFactory.getLogger(MyenergiBaseDeviceHandler.class);
+    private @Nullable ScheduledFuture<?> measurementPollingJob = null;
+
+    protected long serialNumber;
+
+    protected ExpiringCache<Integer> updateThingCache = new ExpiringCache<Integer>(UPDATE_THING_CACHE_TIMEOUT,
+            this::refreshCache);
+
+    public MyenergiBaseDeviceHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing MyEnergiBaseDeviceHandler");
+
+        serialNumber = Long.parseLong(getThing().getUID().getId());
+
+        MyenergiDeviceConfiguration config = getConfigAs(MyenergiDeviceConfiguration.class);
+        if (config.refreshInterval < 10) {
```

Comment:
thing-type.xml allows 5 and greather, please align, whatever value is best.
Also better to extract this to a constant.